src/components/finance/DepositAssets.tsx

```diff
@@ -0,0 +1,289 @@
+/**
+ * This program has been developed by students from the bachelor Computer Science at Utrecht University within the Software Project course.
+ * © Copyright Utrecht University (Department of Information and Computing Sciences)
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useEffect, useState } from 'react';
+import { useDiamondSDKContext } from '@/src/context/DiamondGovernanceSDK';
+import { PREFERRED_NETWORK_METADATA } from '@/src/lib/constants/chains';
+import { NumberPattern } from '@/src/lib/constants/patterns';
+import { parseTokenAmount } from '@/src/lib/utils/token';
+import { BigNumber } from 'ethers';
+import { Controller, useForm, useWatch } from 'react-hook-form';
+import { HiInboxArrowDown } from 'react-icons/hi2';
+import {
+  erc20ABI,
+  useAccount,
+  useContractWrite,
+  useNetwork,
+  usePrepareContractWrite,
+  useWaitForTransaction,
+} from 'wagmi';
+
+import Loading from '../icons/Loading';
+import { Address, AddressLength } from '../ui/Address';
+import { Card } from '../ui/Card';
+import {
+  ConditionalButton,
+  ConnectWalletWarning,
+  Warning,
+} from '../ui/ConditionalButton';
+import { ErrorWrapper } from '../ui/ErrorWrapper';
+import { LabelledInput } from '../ui/Input';
+import { Label } from '../ui/Label';
+import { MainCard } from '../ui/MainCard';
+import {
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectLabel,
+  SelectTrigger,
+  SelectValue,
+} from '../ui/Select';
+
+type Token = (typeof Tokens)[number];
+type DepositAssetsData = {
+  token: Token;
+  amount?: string;
+};
+type AddressString = `0x${string}`;
+type SendData = Record<Token, AddressString | undefined>;
+
+const Tokens = ['Matic', 'SECOIN', 'Other'] as const;
+const tokenAddresses: SendData = {
+  Matic: '0x0000000000000000000000000000000000001010',
+  SECOIN: '0x...',
+  Other: undefined,
+};
+
+export const DepositAssets = ({}) => {
```

Comment:
Can we add a HeaderCard to this page, and possibly add a bac button saying something like "All transfers" similar to the New-proposal and view-proposal pages?